### PR TITLE
Refactor configuration structure

### PR DESCRIPTION
**Pull request by fmessmer**
as discussed with @ipa-mig and @ipa-fmw
this PR leads to changes in 
 - https://github.com/ipa320/cob_driver/pull/332
 - https://github.com/ipa320/cob_manipulation/pull/110
 - https://github.com/ipa320/cob_android/pull/26

additional refactoring is done in:
 - cob_environments https://github.com/ipa320/cob_environments/pull/120
 - cob_navigation https://github.com/ipa320/cob_navigation/pull/85
 - ipa_navigation https://github.com/ipa320/ipa_navigation/pull/184
 - msh https://github.com/ipa320/msh/pull/684

and then testing!

### CODE REVIEW

**Comment by NadiaHG** [comment]
Sorry, where's the advantage here? This just add new subfolders...

**Comment by fmessmer** [comment]
> Sorry, where's the advantage here? This just add new subfolders...

It's sync'ing the structure with the nav configuration and allows for a configuration within a single package, e.g. for demos, while keeping the various config sources grouped semantically without mixing them, i.e. `script_server` configs in its folder, `nav` configs within its folder...

eventually, #437 will introduce a `components` sub-folder beside `robots` (in cob_hardware_config) where we can move component-specific, but robot-independent configs...

**Comment by fmessmer** [comment]
I consider this done so far....
In https://github.com/ipa320/msh/pull/684, I am testing against all related feature branches...

@ipa-mig @ipa-fmw @ipa-nhg 
please review

**Comment by fmessmer** [comment]
maybe we should rebase on top of https://github.com/ipa320/cob_robots/pull/645 after that has been merged...

File: cob_default_robot_config/CMakeLists.txt [hunk 1]
```diff
@@ -10,16 +10,14 @@ if(CATKIN_ENABLE_TESTING)
...
 
 ### INSTALL ###
-foreach(robot ${cob_supported_robots_ROBOTLIST})
-  install(DIRECTORY ${robot}
-    DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION}
-  )
-endforeach()
+install(DIRECTORY robots
+  DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION}
+)
```
Comment (mgruhler) [inline cob_default_robot_config/CMakeLists.txt:20]: This simplification is a nice side effect :-)

**Review by mgruhler — APPROVED** [review]
I'm fine with the changes. However, for functionality, I guess we have to trust in travis :-)

**Review by floweisshardt — APPROVED** [review]

**Comment by NadiaHG** [comment]
I understand the point of unify the configuration but I don't think  this structure is the right one for cob_hardware_config, For this package make not sense, and it is the main configuration package of Care-O-bot. Last time we discussed the structure of cob_robots together,  we decided to move into a new structure where we will only have component/module specific configuration and not robots anymore. And this pull request is actually doing the opposite.

File: cob_moveit_config/cob4-7/.setup_assistant [hunk 2]
```diff
@@ -1,6 +1,6 @@
 moveit_setup_assistant_config:
   URDF:
     package: cob_hardware_config
-    relative_path: cob4-7/urdf/cob4-7.urdf.xacro
+    relative_path: robots/cob4-7/urdf/cob4-7.urdf.xacro
   SRDF:
     relative_path: config/cob4-7.srdf
```
Comment (NadiaHG) [inline cob_moveit_config/cob4-7/.setup_assistant:6]: why do you keep only for cob_moveit_config the structure `package_name/$(arg robot)/...`

**Comment by fmessmer** [comment]
`cob_moveit_config` has been restructured to the same layout for consistency...

**Pull request merged**